trunner/harnesses/micropython.py

```diff
@@ -0,0 +1,221 @@
+#
+# Phoenix-RTOS test runner
+#
+# The harness for the Micropython Test Suite
+#
+# Copyright 2022 Phoenix Systems
+# Authors: Piotr Nieciecki
+#
+
+import re
+import pathlib
+import pexpect
+from .common import TestResult
+
+
+class MicropythonStandardHarness:
+    """Class providing harness for getting test result from micropython test and comparing to the expected output"""
+
+    SPECIAL_TESTS = (
+        "micropython/meminfo.py",
+        "basics/bytes_compare3.py",
+        "basics/builtin_help.py",
+        "thread/thread_exc2.py",
+        "esp32/partition_ota.py",
+        "cmdline/cmd_parsetree.py",
+        "cmdline/cmd_showbc.py",
+        "cmdline/cmd_verbose.py"
+    )
+
+    def is_special_test(path_to_test):
+        p = pathlib.Path(path_to_test)
+        fragments = p.parts
+
+        if fragments[-2] + "/" + fragments[-1] in MicropythonStandardHarness.SPECIAL_TESTS:
+            return True
+        return False
+
+    def get_test_path_and_result(proc):
+        Tools.expect(Tools.PROMPT, proc, timeout=45)
+        program_output = proc.before
+
+        index = program_output.find("\nError:")
+        assert index == -1, f"There was an error during program execution: {program_output}"
+
+        # Looking for a path to the test
+        index = re.search(r'/.+\.py', program_output)
+        assert index is not None, f"Can't find path to the test\nProgram output:\n{program_output}"
+        index = index.span()
+
+        test_path = program_output[index[0]+1:index[1]]
+
+        # After the path there is a program result
+        index = program_output.find('\n')
+        test_result = program_output[index+1:]
+
+        return (test_path, test_result)
+
+    def get_exp_output(test_path, proc):
+        expected_output_path = test_path + ".exp"
+
+        proc.sendline("cat %s\n" % expected_output_path)
+        Tools.expect(Tools.PROMPT, proc)
+
+        # exp_file_cat have a header "cat (path to the expected outpue)"
+        exp_file_cat = proc.before
+
+        # removing the header
+        index = exp_file_cat.find('\n')
+        exp_result = exp_file_cat[index+1:]
+
+        return exp_result
+
+    @staticmethod
+    def harness(p):
+        tests_results = []
+
+        try:
+            (test_path, test_result) = MicropythonStandardHarness.get_test_path_and_result(p)
```

Comment:
You are using methods from the same class, so you do not need to call them by class specificator. You can use:
```suggestion
            (test_path, test_result) = self.get_test_path_and_result(p)
```

That way, if you change the name of that class, you do not need to change all calls :)

**Note:"** applies in more places across both harnesses.

Reply:
I cannot do that, because `harness` is a static method

Reply:
Yeah, i missed that. Sorry. So maybe, to be more pythonic mace those functions as nested functions in harness method? 

I don`t really like the overall idea that harness factory does not return the object, but only static method of some class, but that's whole another topic :)